aggregator/entry.go

```diff
@@ -112,6 +112,53 @@ func (e *Entry) AddMetricWithPoliciesList(
 	mu unaggregated.MetricUnion,
 	pl policy.PoliciesList,
 ) error {
+	switch mu.Type {
+	case unaggregated.BatchTimerType:
+		err := e.writeBatchTimerWithPoliciesList(mu, pl)
+		if mu.BatchTimerVal != nil && mu.TimerValPool != nil {
+			mu.TimerValPool.Put(mu.BatchTimerVal)
+		}
+		return err
+	default:
+		return e.addMetricWithPoliciesList(mu, pl)
+	}
+}
+
+func (e *Entry) writeBatchTimerWithPoliciesList(
+	mu unaggregated.MetricUnion,
+	pl policy.PoliciesList,
+) error {
+	// If there is no limit on the maximum batch size per write, write
+	// all timers at once.
+	maxTimerBatchSizePerWrite := e.opts.MaxTimerBatchSizePerWrite()
```

Comment:
This optimization seems a little unintuitive to me because we're going to have to acquire the locks we need multiple times if the number of timer values is greater than the max batch timer size. So even though other calls to `AddMetricWithPoliciesList` may have to wait less time to acquire the locks they need, calls with large batches will take longer since they have to acquire the locks multiple times. I suppose there might an equilibrium point for the percentage of calls with large batches below which it makes sense to have a max batch timer size and above which it does not. It might be interesting to do an experiment with the agg tier where we set half the cluster to use a max batch timer size and the other half to set it to zero for an extended period of time so we can compare the two.